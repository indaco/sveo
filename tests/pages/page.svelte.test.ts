import { describe, test, expect } from 'vitest';
import '@testing-library/jest-dom/vitest';
import { render, screen } from '@testing-library/svelte';
import Page from '../../src/routes/+page.svelte';

describe('/+page.svelte', () => {
	test('should render h1', () => {
		render(Page);
		expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('@indaco/sveo');
	});

	test('sets <title> from homePage', () => {
		render(Page);
		expect(document.title).toBe('Home Page');
	});

	test('sets meta description from homePage', () => {
		render(Page);
		const meta = document.head.querySelector('meta[name="description"]');
		expect(meta).toHaveAttribute('content', 'This is the description for the Home Page');
	});

	test('sets og:type to Website', () => {
		render(Page);
		const meta = document.head.querySelector('meta[property="og:type"]');
		expect(meta).toHaveAttribute('content', 'website');
	});

	test('sets og:title', () => {
		render(Page);
		const meta = document.head.querySelector('meta[property="og:title"]');
		expect(meta).toHaveAttribute('content', 'Home Page');
	});

	test('sets og:description', () => {
		render(Page);
		const meta = document.head.querySelector('meta[property="og:description"]');
		expect(meta).toHaveAttribute('content', 'This is the description for the Home Page');
	});

	test('sets twitter:card to summary', () => {
		render(Page);
		const meta = document.head.querySelector('meta[property="twitter:card"]');
		expect(meta).toHaveAttribute('content', 'summary');
	});

	test('renders JSON-LD WebSite', () => {
		render(Page);
		const script = document.head.querySelector(
			'script[type="application/ld+json"][data-testid="jsonld-website"]'
		);
		expect(script?.textContent).toContain('"@type": "WebSite"');
		expect(script?.textContent).toContain('"name": "example.com"');
		expect(script?.textContent).toContain('"inLanguage": "en-GB"');
	});

	test('renders JSON-LD WebPage', () => {
		render(Page);
		const script = document.head.querySelector(
			'script[type="application/ld+json"][data-testid="jsonld-webpage"]'
		);
		expect(script?.textContent).toContain('"@type": "WebPage"');
		expect(script?.textContent).toContain('"name": "Home Page"');
	});

	test('renders JSON-LD SiteNavigationElements', () => {
		render(Page);
		const script = document.head.querySelector(
			'script[type="application/ld+json"][data-testid="jsonld-sitenavigationelements"]'
		);
		expect(script?.textContent).toContain('"@type": "SiteNavigationElement"');
		expect(script?.textContent).toContain('"name": "Home"');
		expect(script?.textContent).toContain('"url": "https://example.com/about"');
	});

	test('renders correct <head> snapshot', () => {
		render(Page);
		expect(document.head.innerHTML).toMatchSnapshot();
	});
});
