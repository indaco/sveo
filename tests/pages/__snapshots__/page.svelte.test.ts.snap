// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`/+page.svelte > renders correct <head> snapshot 1`] = `
"<title>Home Page</title><link rel="canonical" href="https://example.com"> <meta name="description" content="This is the description for the Home Page"><!----> <meta name="keywords" content="sveltekit, components, tests, jest"><!----> <!----><meta property="og:type" content="website"> <meta property="og:url" content="https://example.com"> <meta property="og:title" content="Home Page"> <meta property="og:description" content="This is the description for the Home Page"><!----> <!----> <!----><!----> <!----><!----><meta property="twitter:card" content="summary"> <meta property="twitter:title" content="Home Page"> <!----> <meta property="twitter:description" content="This is the description for the Home Page"><!----> <!----> <!----> <!----><!----><!----><script type="application/ld+json" data-testid="jsonld-website">{
  "@context": "https://schema.org",
  "@type": "WebSite",
  "@id": "https://example.com/#website",
  "name": "example.com",
  "url": "https://example.com",
  "description": "Example.com is your perfect fit.",
  "keywords": [
    "sveltekit, components, tests, jest"
  ],
  "inLanguage": "en-GB",
  "license": "MIT License @ 2021-present example.com",
  "sameAs": [
    "https://github.com/examplecom"
  ],
  "creator": {
    "@type": "Person",
    "name": "username",
    "email": "github@example.com",
    "url": "https://yourwebgarder.com",
    "address": {
      "@type": "PostalAddress",
      "addressLocality": "City",
      "streetAddress": "somewhere 101"
    },
    "jobTitle": "webmaster"
  }
}</script><!----><script type="application/ld+json" data-testid="jsonld-sitenavigationelements">{
  "@context": "https://schema.org",
  "@type": "ItemList",
  "itemListElement": [
    {
      "@type": "SiteNavigationElement",
      "position": 1,
      "name": "Home",
      "url": "https://example.com/"
    },
    {
      "@type": "SiteNavigationElement",
      "position": 2,
      "name": "About",
      "url": "https://example.com/about"
    },
    {
      "@type": "SiteNavigationElement",
      "position": 3,
      "name": "GitHub",
      "url": "https://github.com/indaco"
    }
  ]
}</script><!----><script type="application/ld+json" data-testid="jsonld-webpage">{
  "@context": "https://schema.org",
  "@type": "WebPage",
  "name": "Home Page",
  "description": "This is the description for the Home Page",
  "keywords": [
    "sveltekit, components, tests, jest"
  ]
}</script>"
`;
